mitiq/folding_pyquil.py

```diff
@@ -41,7 +41,7 @@ def fold_gates(circuit: Program, sub_indices: List[int]) -> Program:
     """Applies the map G -> G G^dag G to a subset of gates of the input circuit
     determined by sub_indices."""
 
-    out = Program()
+    out = circuit.copy_everything_except_instructions()
```

Comment:
neat find! I didn't know they already had that function :)

Reply:
Also wasn't aware of this -- nice!